cdap-messaging-ext-spanner/src/main/java/io/cdap/cdap/messaging/spanner/SpannerUtil.java

```diff
@@ -86,10 +94,31 @@ static Credentials getCredentials(Map<String, String> cConf) throws IOException
     return null;
   }
 
-  public static Spanner getSpannerService(String projectID, Credentials credentials) {
+  /**
+   * Obtains a Spanner service client, configuring it for emulator if {@code SPANNER_EMULATOR_HOST}
+   * system property is set.
+   */
+  public static Spanner getSpannerService(Map<String, String> conf, String projectID,
+      @Nullable Credentials credentials) {
     SpannerOptions.Builder builder = SpannerOptions.newBuilder().setProjectId(projectID);
-    if (credentials != null) {
+
+    String emulatorHost = conf.get(SPANNER_EMULATOR_HOST);
+    if (emulatorHost != null && !emulatorHost.isEmpty()) {
```

Comment:
can we use `!Strings.isNullOrEmpty()`? Do we have guava in the extension? If not, we can keep it as is.

Not sure why this property is present : https://github.com/cdapio/cdap/blob/5b1cce39f2befc301a3741a6cdaf43ed7e611b25/cdap-messaging-ext-spanner/pom.xml#L31C6-L31C19